client/src/pages/admin/MergeLocations.js

```diff
@@ -0,0 +1,516 @@
+import { Button, Callout, Tooltip } from "@blueprintjs/core"
+import { DEFAULT_PAGE_PROPS, DEFAULT_SEARCH_PROPS } from "actions"
+import API from "api"
+import { gql } from "apollo-boost"
+import { LocationOverlayRow } from "components/advancedSelectWidget/AdvancedSelectOverlayRow"
+import AdvancedSingleSelect from "components/advancedSelectWidget/AdvancedSingleSelect"
+import ApprovalSteps, {
+  PLANNING_APPROVAL,
+  PUBLICATION_APPROVAL
+} from "components/approvals/ApprovalSteps"
+import Leaflet from "components/Leaflet"
+import Messages from "components/Messages"
+import { GRAPHQL_NOTES_FIELDS } from "components/Model"
+import {
+  jumpToTop,
+  mapPageDispatchersToProps,
+  PageDispatchersPropType,
+  useBoilerplate
+} from "components/Page"
+import _escape from "lodash/escape"
+import { Location } from "models"
+import GeoLocation from "pages/locations/GeoLocation"
+import PropTypes from "prop-types"
+import React, { useMemo, useState } from "react"
+import { connect } from "react-redux"
+import { useHistory } from "react-router-dom"
+import { toast } from "react-toastify"
+import LOCATIONS_ICON from "resources/locations.png"
+
+const locationFilters = {
+  allLocations: {
+    label: "All locations",
+    queryVars: {}
+  }
+}
+
+const GQL_MERGE_LOCATION = gql`
+  mutation(
+    $loserLocationUuid: String!
+    $loserLocationUuid2: String!
+    $location: LocationInput!
+  ) {
+    mergeLocation(
+      loserLocationUuid: $loserLocationUuid
+      loserLocationUuid2: $loserLocationUuid2
+      location: $location
+    ) {
+      uuid
+    }
+  }
+`
+
+const selectFields = `
+  uuid
+  name
+  lat
+  lng
+  status
+  planningApprovalSteps {
+    uuid
+    name
+    approvers {
+      uuid
+      name
+      person {
+        uuid
+        name
+        rank
+        role
+        avatar(size: 32)
+      }
+    }
+  }
+  approvalSteps {
+    uuid
+    name
+    approvers {
+      uuid
+      name
+      person {
+        uuid
+        name
+        rank
+        role
+        avatar(size: 32)
+      }
+    }
+  }
+  ${GRAPHQL_NOTES_FIELDS}
+`
+
+const LocationColumn = ({
+  location,
+  setLocation,
+  setFieldValue,
+  reversed,
+  style
+}) => {
+  const map = useMemo(() => {
```

Comment:
There shouldn't be a need to memoize the map component. Sometimes we memoize the markers, but also not needed in this case as it is a single one

Reply:
I used it for performance optimization by avoiding unnecessary renders on Leaflet, but it works just fine without it. So I will remove it.